uptimerobot/api/alert_contact.go

```diff
@@ -94,12 +144,12 @@ func (client UptimeRobotApiClient) CreateAlertContact(req AlertContactCreateRequ
 		return
 	}
 
-	return client.GetAlertContact(int(alertcontact["id"].(float64)))
+	return client.GetAlertContact(alertcontact["id"].(string))
```

Comment:
The alertcontact["id"] here is seems to be a float64 type which can't be converted by .(string).

Reply:
According to the API documentation the ID is a string https://uptimerobot.com/api

![image](https://user-images.githubusercontent.com/365827/55827628-ef0e4080-5b0a-11e9-9d2b-e552d5d71f14.png)

I stumbled upon this because one of the ID's generated for me by the API started with a `0` which would be stripped when converting form a string to a int/float64 and thus Terraform would record an invalid ID in the state file.

Reply:
May be this is a uptimerobot issure. At least for me, the response of this API is different from the document.
https://gist.github.com/leeif/0bc4eb5f9ca03b9c487169d98167cc40

Reply:
I think this is an issue at UR's end.

Alert contact create returns a integer:

```
$ curl -sS -X POST -H "Cache-Control: no-cache" -H "Content-Type: application/x-www-form-urlencoded" -d "api_key=XXXX&format=json&friendly_name=Test2&type=2&value=test2%40example.com"  "https://api.uptimerobot.com/v2/newAlertContact"|jq .
{
  "stat": "ok",
  "alertcontact": {
    "id": 2809628
  }
}
```

While getAlertContacts returns a string:
```
$ curl -sS -X POST -H "Cache-Control: no-cache" -H "Content-Type: application/x-www-form-urlencoded" -d "api_key=XXXXXX&format=json"  "https://api.uptimerobot.com/v2/getAlertContacts"|jq .
{
  "stat": "ok",
  "offset": 0,
  "limit": 50,
  "total": 3,
  "alert_contacts": [
    {
      "id": "0717807",
      "friendly_name": "uptimerobottest@ijohan.nl",
      "type": 2,
      "status": 2,
      "value": "uptimerobottest@ijohan.nl"
    },
    {
      "id": "2806958",
      "friendly_name": "Test",
      "type": 2,
      "status": 0,
      "value": "test@example.com"
    },
    {
      "id": "2809628",
      "friendly_name": "Test2",
      "type": 2,
      "status": 0,
      "value": "test2@example.com"
    }
  ]
}
```

One thing I do notice is that the ID of the default alert contact starts with 0 but the others are in a completely different number range.

Reply:
Also querying without the zero does not work:
```
$ curl -sS -X POST -H "Cache-Control: no-cache" -H "Content-Type: application/x-www-form-urlencoded" -d "api_key=XXXXX&format=json&alert_contacts=717807"  "https://api.uptimerobot.com/v2/getAlertContacts"|jq .
{
  "stat": "fail",
  "error": {
    "type": "not_found",
    "message": "Alert Contact(id: 717807) not found."
  }
}
```

```
$ curl -sS -X POST -H "Cache-Control: no-cache" -H "Content-Type: application/x-www-form-urlencoded" -d "api_key=XXXXX&format=json&alert_contacts=0717807"  "https://api.uptimerobot.com/v2/getAlertContacts"|jq .
{
  "stat": "ok",
  "offset": 0,
  "limit": 50,
  "total": 1,
  "alert_contacts": [
    {
      "id": "0717807",
      "friendly_name": "uptimerobottest@ijohan.nl",
      "type": 2,
      "status": 2,
      "value": "uptimerobottest@ijohan.nl"
    }
  ]
}
```

Reply:
So I added a conversion to string instead of casting. I don't think UR will change their API and break all clients out there (didn't attempt to contact them though). But this should cover both cases:
- in `newAlertContact` where the ID is returned as a integer
- in all other cases where it is handled as a string

Also only the response values are types (due to JSON) and the request values not (as they are HTTP parameters).